spectrochempy/analysis/preprocessing/baseline.py

```diff
@@ -0,0 +1,899 @@
+# -*- coding: utf-8 -*-
+# ======================================================================================
+# Copyright (©) 2015-2023 LCS - Laboratoire Catalyse et Spectrochimie, Caen, France.
+# CeCILL-B FREE SOFTWARE LICENSE AGREEMENT
+# See full LICENSE agreement in the root directory.
+# ======================================================================================
+"""
+This module implements the `Baseline` class for baseline corrections and related methods.
+"""
+
+import numpy as np
+import scipy.interpolate
+import scipy.signal
+import traitlets as tr
+from scipy import sparse
+from scipy.sparse.linalg import spsolve
+
+from spectrochempy.analysis._base import (
+    AnalysisConfigurable,
+    NotFittedError,
+    _wrap_ndarray_output_to_nddataset,
+)
+from spectrochempy.analysis.preprocessing.utils import lls, lls_inv
+from spectrochempy.application import info_, warning_
+from spectrochempy.core.processors.concatenate import concatenate
+from spectrochempy.utils.coordrange import trim_ranges
+from spectrochempy.utils.decorators import deprecated, signature_has_configurable_traits
+from spectrochempy.utils.docstrings import _docstring
+from spectrochempy.utils.misc import TYPE_FLOAT, TYPE_INTEGER
+from spectrochempy.utils.traits import NDDatasetType
+
+__all__ = [
+    "Baseline",
+    "abc",
+    "basc",
+    "detrend",
+    "asls",
+    "snip",
+    "lls",
+    "lls_inv",
```

Comment:
- should "lls" and "lls_inv" be make public ?, if yes, why not in utils.py where they are defined  ?